### ISSUE TITLE
Support Kotlin multi-platform

### ISSUE THREAD

**Issue by rocketraman** [apache/logging-log4j-kotlin#59]
It would be nice if the API supported Kotlin multi-platform, and so could be used on other platforms that Kotlin targets, such as native, JavaScript, and WASM.

Log4j2 is obviously a JVM-only solution but this library is mostly just an API. We could make it easier for users to plugin in different backends for different platforms, and provide some reasonable choices out of the box for each platform.

**Comment by jvz** [comment]
Yes, I like this idea. Once that is in place, we can even implement some minimal multi-platform implementations (e.g., there's a common `println` and `readln` function for all platforms), and we may even be able to support some OS-native logging APIs (like syslog, systemd, Windows event logger, and whatever the macOS API is).

**Comment by StefanOltmann** [comment]
Plan of action: Start by something like making a _slf4j_ Kotlin Multiplatform port… name it _slf4k_. Then be the first library that implements it and convince others like Kermit to do so, too. Then offer JetBrains to include slf4k in the stdlib by the time most logger implantations support it.
